runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,887 @@
+                    + " Please check the resource in AWS console, delete relevant resource(s) "
+                    + "if necessary, and re-run your Runhouse code."
+                )
+            time.sleep(1)
+            lambdas_log_groups = self._get_log_group_names(log_group_prefix)
+            time_passed += 1
+        yield True
+
+    def _invoke(self, *args, **kwargs) -> Any:
+        log_group_prefix = "/aws/lambda/"
+        lambdas_log_groups = self._get_log_group_names(log_group_prefix)
+        curr_lambda_log_group = f"{log_group_prefix}{self.name}"
+        invoke_for_the_first_time = curr_lambda_log_group not in lambdas_log_groups
+
+        payload_invoke = {}
+
+        if len(args) > 0 and self.args_names is not None:
+            payload_invoke = {self.args_names[i]: args[i] for i in range(len(args))}
+        invoke_res = self.lambda_client.invoke(
+            FunctionName=self.name,
+            Payload=json.dumps({**payload_invoke, **kwargs}),
+            LogType="Tail",
+        )
+        return_value = invoke_res["Payload"].read().decode("utf-8")
+
+        try:
+            logger.error(invoke_res["FunctionError"])
+            return_value = json.loads(return_value)
+            raise RuntimeError(
+                f"Failed to run {self.name}. Error: {return_value['errorType']}, {return_value['errorMessage']}"
+            )
+
+        except KeyError:
+            if invoke_for_the_first_time:
+                lambdas_log_groups = self._get_log_group_names(log_group_prefix)
+                with self._wait_till_log_group_is_created(
+                    curr_lambda_log_group, lambdas_log_groups, log_group_prefix
+                ):
+                    self.logs_client.put_retention_policy(
+                        logGroupName=curr_lambda_log_group,
+                        retentionInDays=self.retention_time,
+                    )
+
+            log_lines = "Function Logs are:\n" + base64.b64decode(
+                invoke_res["LogResult"]
+            ).decode("utf-8")
+
+            print(log_lines)
+            return return_value
+
+    def map(self, *args, **kwargs):
+        """Map a function over a list of arguments.
+
+        Example:
+            >>> # The my_lambda_handler.py file
+            >>> def my_summer(arg1, arg2, arg3):
+            >>>     return arg1 + arg2 + arg3
+            >>>
+            >>> # your 'main' python file, where you are using runhouse
+            >>> my_lambda = rh.lambda_function(path_to_code=["full/path/to/my_lambda_handler.py"],
+            >>> handler_function_name='my_summer',
+            >>> runtime='python_3_9',
+            >>> name="my_summer").to()
+            >>> output = my_lambda.map([1, 2], [1, 4], [2, 3])
+            >>> # output = ["4", "9"] (It returns str type because of AWS API)
+
+        """
+
+        return [self._invoke(*args, **kwargs) for args in zip(*args)]
+
+    #
+    def starmap(self, args_lists, **kwargs):
```

Comment:
is `map` and `starmap` working again? otherwise should exclude those in the docs

Reply:
They always worked, Lambda implementation overrides function implementation.